tests/test_fock.py

```diff
@@ -625,6 +627,112 @@ def circuit(n, a):
+
+        res = circuit(a, phi)
+        assert np.allclose(res, ref_probs, atol=tol, rtol=0)
+
+    def test_multi_mode_probability(self, tol):
+        """Test that a product of coherent states returns the correct probability"""
+        cutoff = 10
+        dev = qml.device("strawberryfields.fock", wires=2, cutoff_dim=cutoff)
+
+        @qml.qnode(dev)
+        def circuit(a, phi):
+            qml.Displacement(a, phi, wires=0)
+            qml.Displacement(a, phi, wires=1)
+            return qml.probs(wires=[0, 1])
+
+        a = 0.4
+        phi = -0.12
+
+        alpha = a * np.exp(1j * phi)
+        n = np.arange(cutoff)
+        ref_probs = np.abs(np.exp(-0.5 * np.abs(alpha) ** 2) * alpha ** n / np.sqrt(fac(n))) ** 2
+        ref_probs = np.kron(ref_probs, ref_probs)
+
+        res = circuit(a, phi)
+        assert np.allclose(res, ref_probs, atol=tol, rtol=0)
+
+    def test_marginal_probability(self, tol):
+        """Test that a coherent state marginal probability is correct"""
+        cutoff = 10
+        dev = qml.device("strawberryfields.fock", wires=2, cutoff_dim=cutoff)
+
+        @qml.qnode(dev)
+        def circuit(a, phi):
+            qml.Displacement(a, phi, wires=1)
+            return qml.probs(wires=1)
+
+        a = 0.4
+        phi = -0.12
+
+        alpha = a * np.exp(1j * phi)
+        n = np.arange(cutoff)
+        ref_probs = np.abs(np.exp(-0.5 * np.abs(alpha) ** 2) * alpha ** n / np.sqrt(fac(n))) ** 2
+
+        res = circuit(a, phi)
+        assert np.allclose(res, ref_probs, atol=tol, rtol=0)
+
+    def test_finite_diff_coherent(self, tol):
+        """Test that the jacobian of the probability for a coherent states is
+        approximated well with finite differences"""
+        cutoff = 10
+        dev = qml.device("strawberryfields.fock", wires=4, cutoff_dim=cutoff)
+
+        def circuit(a, phi):
+            qml.Displacement(a, phi, wires=0)
+            return qml.probs(wires=[0])
+
+        tol = 7*1e-2
+        a = 0.4
+        phi = -0.12
+
+        alpha = a * np.exp(1j * phi)
+        n = np.arange(cutoff)
+
+        circuit = qml.QNode(circuit, dev)
+        t1 = 2*(a**2)**(-0.5 + n)
+        t2 = np.exp(-(a**2) + 1j*(phi + (-1 + n)*phi))
+        t3 = np.sqrt((a*np.exp(1j*phi))**(2*n))
+        t4 = n-a*np.sqrt(a**2)*np.sign(a)
+        denom = (a*np.exp(1j*phi))**n*fac(n)
+        diff_a = t1*t2*t3*t4/denom
+
+        tol = 7*1e-2
+
+        t1 = 2j*(a**2)**(-0.5 + n)
+        t2 = np.exp(-(a**2) + 1j*(-1 + n)*phi)
+        t3 = (a*np.exp(1j*phi))**(1-n)
+        t4 = np.sqrt((a*np.exp(1j*phi))**(2*n))
+        t5 = n
+        denom = fac(n)
+        partial_phi = t1*t2*t3*t4*t5/denom
+
+        params = [a, phi]
+        res_F = circuit.jacobian(params, method="F")
+        assert np.allclose(res_F[:,0], diff_a, atol=tol, rtol=0)
+        #assert np.allclose(res_F[:,1], partial_phi, atol=tol, rtol=0)
```

Comment:
@josh146 something funky seems to be going on here :thinking:

Reply:
Have pushed a commit to fix the analytic formula here :)